core/node/consensus/src/storage/store.rs

```diff
@@ -175,14 +178,23 @@ impl StoreRunner {
                 // Loop updating `blocks_persisted` whenever blocks get pruned.
                 const POLL_INTERVAL: time::Duration = time::Duration::seconds(1);
                 loop {
-                    let range = pool
-                        .connection(ctx)
-                        .await?
-                        .block_certificates_range(ctx)
-                        .await
-                        .wrap("block_certificates_range()")?;
-                    blocks_persisted.update(range);
-                    ctx.sleep(POLL_INTERVAL).await?;
+                    let span = tracing::info_span!("blocks_persisted_loop");
```

Comment:
ditto, you can put it on the outer `async { ... }` instead

Reply:
Correct me if I am wrong, but this way the span would never be exited. The loop spans are ugly but I am unsure on how to achieve something similar with less invasive approach.

Reply:
Yup, the span must be within the loop.

I'm also concerned by the somewhat intrusive spans in the code, but I think if something is interesting for us _as a span_ then it probably deserves its own function.

Reply:
That's a fair point and I would be glad to refactor loop body into a function if @pompon0 agrees